util/network/networking.go

```diff
@@ -0,0 +1,92 @@
+package network
+
+import (
+	"fmt"
+
+	"github.com/moby/buildkit/identity"
+	"github.com/vishvananda/netlink"
+)
+
+//NetworkInterface for workers
```

Comment:
Add also 
```
type NetworkProvider interface {
   NewInterface() (NetworkInterface, error)
}
```
and this is the one that is passed to the worker constructor.

This will probably change when we add a way to control between host and bridge but should be good enough for current iteration.